### PR TITLE
fix: properly set aria attributes on components

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #10407

## Summary

- only sets `aria-expanded` on button when present on host
- properly sets `boolean` values to strings on other components 
- removes `aria-hidden` from screen reader text

### CODE REVIEW

File: packages/calcite-components/src/components/table-row/table-row.tsx [hunk 1]
```diff
@@ -410,7 +414,7 @@ export class TableRow implements InteractiveComponent, LocalizedComponent {
         <InteractiveContainer disabled={this.disabled}>
           <tr
             aria-rowindex={this.positionAll + 1}
-            aria-selected={this.selected}
+            aria-selected={toAriaBoolean(this.selected)}
```
Comment (driskull) [inline packages/calcite-components/src/components/table-row/table-row.tsx:417]: This wasn't doing anything. This fixes it.

File: packages/calcite-components/src/components/table-header/table-header.tsx [hunk 2]
```diff
@@ -261,11 +261,7 @@ export class TableHeader implements LocalizedComponent, LoadableComponent, T9nCo
             />
           )}
           {(this.selectionCell || this.numberCell) && (
-            <span
-              aria-hidden={true}
-              aria-live={this.focused ? "polite" : "off"}
-              class={CSS.assistiveText}
-            >
+            <span aria-live={this.focused ? "polite" : "off"} class={CSS.assistiveText}>
```
Comment (driskull) [inline packages/calcite-components/src/components/table-header/table-header.tsx:264]: screen reader text shouldn't have aria-hidden.

File: packages/calcite-components/src/components/table-cell/table-cell.tsx [hunk 3]
```diff
@@ -243,11 +243,7 @@ export class TableCell
             tabIndex={staticCell ? -1 : 0}
           >
             {(this.selectionCell || this.readCellContentsToAT) && (
-              <span
-                aria-hidden={true}
-                aria-live={this.focused ? "polite" : "off"}
-                class={CSS.assistiveText}
-              >
+              <span aria-live={this.focused ? "polite" : "off"} class={CSS.assistiveText}>
```
Comment (driskull) [inline packages/calcite-components/src/components/table-cell/table-cell.tsx:246]: screen reader text shouldn't have aria-hidden.

File: packages/calcite-components/src/components/button/button.tsx [hunk 4]
```diff
@@ -267,7 +267,7 @@ export class Button
       <InteractiveContainer disabled={this.disabled}>
         <Tag
           aria-busy={toAriaBoolean(this.loading)}
-          aria-expanded={this.el.ariaExpanded}
+          aria-expanded={this.el.ariaExpanded ? this.el.ariaExpanded : null}
```
Comment (driskull) [inline packages/calcite-components/src/components/button/button.tsx:270]: only set aria-expanded when set on host

File: packages/calcite-components/src/components/text-area/text-area.tsx [hunk 5]
```diff
@@ -366,7 +366,7 @@ export class TextArea
           </footer>
           <HiddenFormInputSlot component={this} />
           {this.isCharacterLimitExceeded() && (
-            <span aria-hidden={true} aria-live="polite" class={CSS.assistiveText} id={this.guid}>
+            <span aria-live="polite" class={CSS.assistiveText} id={this.guid}>
```
Comment (driskull) [inline packages/calcite-components/src/components/text-area/text-area.tsx:369]: screen reader text shouldn't have aria-hidden.

File: packages/calcite-components/src/components/tree/tree.tsx [hunk 6]
```diff
@@ -88,9 +88,9 @@ export class Tree {
         aria-multiselectable={
           this.child
             ? undefined
-            : (
-                this.selectionMode === "multiple" || this.selectionMode === "multichildren"
-              ).toString()
+            : toAriaBoolean(
```
Comment (driskull) [inline packages/calcite-components/src/components/tree/tree.tsx:91]: 🧹

**Review by jcfranco — APPROVED** [review]
LGTM! 

Before merging, can we create and link an issue for this, so it can go through the verification process?

**Comment by driskull** [comment]
Created and linked issue

**Pull request merged**